editor/ok_interface.py

```diff
@@ -93,116 +30,167 @@ def write(self, message):
     def flush(self):
         sys.__stdout__.flush()
 
-
-def run_tests():
-    LOGGING_FORMAT = '%(levelname)s  | %(filename)s:%(lineno)d | %(message)s'
-    logging.basicConfig(format=LOGGING_FORMAT)
-    log = logging.getLogger('client')  # Get top-level logger
-
-    # noinspection PyUnresolvedReferences
-    from client.cli.ok import parse_input
-    log.setLevel(logging.ERROR)
-
-    args = parse_input(["--all", "--verbose", "--local"])
-
+def capture_output(console, lines):
     old_stdout = sys.stdout
     sys.stdout = out = PrintCapture()
+    result = console._interpret_lines(lines)
+    sys.stdout = old_stdout
+    return result, out.log
 
-    assign = assignment.load_assignment(None, args)
+class PromptOutput(metaclass=ABCMeta):
+    @abstractmethod
+    def represenation(self):
+        pass
 
-    msgs = messages.Messages()
-    for name, proto in assign.protocol_map.items():
-        log.info('Execute {}.run()'.format(name))
-        proto.run(msgs)
+@dataclass
+class AreDifferent(PromptOutput):
+    prompt: str
+    expected: str
+    actual: str
+    def represenation(self):
+        return "{prompt}\n{expected}\n{actual}".format(
+            prompt=self.prompt,
+            expected=pad("; expected: ", ";", self.expected),
+            actual  =pad("; actual  : ", ";", self.actual)
+        )
 
-    msgs['timestamp'] = str(datetime.now())
+@dataclass
+class Same(PromptOutput):
+    prompt: str
+    output: str
+    def represenation(self):
+        return "{prompt}\n{output}".format(
+            prompt=self.prompt,
+            output=pad("; success: ", ";", self.output)
+        )
+
+class TestCaseResult(metaclass=ABCMeta):
+    @abstractmethod
+    def success(self):
+        pass
+
+    @abstractmethod
+    def output(self):
+        pass
+
+    @property
+    def dictionary(self):
+        return {
+            "code" : self.output(),
+            "passed" : self.success()
+        }
 
-    if assign:
-        assign.dump_tests()
+@dataclass
+class FailureInSetup(TestCaseResult):
+    setup_out: str
```

Comment:
```suggestion
    setup_out: List[str]
```
Not sure, but is this a str or List[str]? You're joining it later on, right?

Reply:
well this is no longer annotated, so resolved ;-)